docroot/modules/humsci/hs_field_helpers/src/Plugin/views/query/Sql.php

```diff
@@ -0,0 +1,122 @@
+
+/**
+ * Class Sql that adds extra functionality to views Sql.
+ *
+ * @package Drupal\hs_field_helpers\Plugin\views\query
+ */
+class Sql extends OriginalSql {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAggregationInfo() {
+    $info = parent::getAggregationInfo();
+    $info['greatest'] = [
+      'title' => $this->t('Greatest'),
+      'method' => 'aggregationMethodSimple',
+      'handler' => [
+        'argument' => 'groupby_numeric',
+        'field' => 'numeric',
+        'filter' => 'groupby_numeric',
+        'sort' => 'groupby_numeric',
+      ],
+    ];
+    $info['least'] = [
+      'title' => $this->t('Least'),
+      'method' => 'aggregationMethodSimple',
+      'handler' => [
+        'argument' => 'groupby_numeric',
+        'field' => 'numeric',
+        'filter' => 'groupby_numeric',
+        'sort' => 'groupby_numeric',
+      ],
+    ];
+    return $info;
+  }
+
+  /**
+   * Alter the view query in consolidate multiple date field sorting.
+   *
+   * @param \Drupal\views\ViewExecutable $view
+   *   View object for the query.
+   */
+  public function alterQuery(ViewExecutable $view) {
+    $date_fields = $this->getDateFields();
+
+    // Not enough date fields to care about altering.
+    if (!(count($date_fields) > 1)) {
+      return;
+    }
+
+    $expression = [];
+    $orderby_delta = -1;
+
+    // Build the expression.
+    foreach ($date_fields as $field_alias => $direction) {
+      $field = $this->fields[$field_alias];
+
+      // Set null values to 0.
+      $expression[] = 'COALESCE(' . $field['table'] . '.' . $field['field'] . ', 0)';
+
+      // Unset the original sorting.
+      foreach ($this->orderby as $delta => $order_item) {
+        if ($order_item['field'] == $field_alias) {
+          $orderby_delta = $delta;
+          unset($this->orderby[$delta]);
+        }
+      }
+    }
+
+    // Add a new field to the select query and use an expression.
+    $this->fields['max_date'] = [
+      'function' => 'greatest',
+      'field' => implode(', ', $expression),
+      'alias' => 'max_date',
+    ];
+
+    // New sorting with our expression above.
+    $this->orderby[$orderby_delta] = [
+      'field' => 'max_date',
+      'direction' => $direction,
+    ];
+
+    // Fix the order weights based on the deltas.
+    ksort($this->orderby);
+    $this->orderby = array_values($this->orderby);
+  }
+
+  /**
+   * Get an array of date fields being used as sorting.
+   *
+   * @return array
+   *   Keyed array of field alias and sort direction.
+   */
+  protected function getDateFields() {
+    $field_types = ['datetime', 'daterange'];
+    $date_fields = [];
+
+    foreach ($this->orderby as $order_item) {
+      $field_alias = $order_item['field'];
+      $field_data = $this->fields[$field_alias];
+      $table = $field_data['table'];
+      if (strpos($table, '__') === FALSE) {
+        continue;
+      }
+      list($entity_type, $field_name) = explode('__', $table);
```

Comment:
http://php.net/manual/en/function.list.php

The list function's behavior is different in PHP5 vs PHP7. Is there a way you can ensure that this module is dependant on PHP7 or protect against running backward on PHP5.6

Reply:
I read that warning and i dont believe it applies to this code. I tried out different versions using http://sandbox.onlinephpfunctions.com/
php 5 and php 7 both produce the same result when exploding a string.